elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryThread.java

```diff
@@ -112,30 +98,33 @@ else if (queryObj.getQueryType().equals(QueryType.GRAPHQL_V1_0)) {
     /**
      * This method parses the url and gets the query params.
      * And retrieves path to be used by underlying Elide.get method
-     * @param query query from the Async request
+     * @param uri URIBuilder instance
      * @throws URISyntaxException URISyntaxException from malformed or incorrect URI
      * @return Path extracted from URI
      */
-    protected String getPath(String query) throws URISyntaxException {
-        URIBuilder uri;
-        uri = new URIBuilder(query);
+    private String getPath(URIBuilder uri) {
         return uri.getPath();
     }
 
-    private ElideResponse executeJsonApiRequest(AsyncQuery queryObj, UUID requestId) throws URISyntaxException {
-        MultivaluedMap<String, String> queryParams = getQueryParams(queryObj.getQuery());
+    private ElideResponse executeJsonApiRequest(Elide elide, User user, String apiVersion) throws URISyntaxException {
+        UUID requestUUId = UUID.fromString(queryObj.getRequestId());
+        URIBuilder uri = new URIBuilder(queryObj.getQuery());
+        MultivaluedMap<String, String> queryParams = getQueryParams(uri);
         log.debug("Extracted QueryParams from AsyncQuery Object: {}", queryParams);
 
         //TODO - we need to add the baseUrlEndpoint to the queryObject.
-        ElideResponse response = elide.get("", getPath(queryObj.getQuery()), queryParams, user, apiVersion, requestId);
+        ElideResponse response = elide.get("", getPath(uri), queryParams, user, apiVersion, requestUUId);
         log.debug("JSONAPI_V1_0 getResponseCode: {}, JSONAPI_V1_0 getBody: {}",
                 response.getResponseCode(), response.getBody());
         return response;
     }
 
-    private ElideResponse executeGraphqlRequest(AsyncQuery queryObj, UUID requestId) throws URISyntaxException {
+    private ElideResponse executeGraphqlRequest(Map<String, QueryRunner> runners, User user, String apiVersion)
+            throws URISyntaxException {
+        UUID requestUUId = UUID.fromString(queryObj.getRequestId());
```

Comment:
Any reason we moved this logic (requestUUId assignment) inside each executeX method?